src/upload/system/library/transbank/utils/LogHandler.php

```diff
@@ -0,0 +1,158 @@
+    {
+        $bytes = @filesize($path);
+        if ($bytes > 0) {
+            $unit = intval(log($bytes, 1024));
+            $units = array('B', 'KB', 'MB', 'GB');
+            if (array_key_exists($unit, $units) === true) {
+                return sprintf('%d %s', $bytes / pow(1024, $unit), $units[$unit]);
+            }
+        }
+        return '0 B';
+    }
+
+    /**
+     * Create the log directory if it does not exist.
+     */
+    private function setMakeLogDir(): void
+    {
+        try {
+            if (!file_exists($this->logDir)) {
+                mkdir($this->logDir, 0777, true);
+            }
+        } catch (\Exception $e) {
+            error_log('Error al crear la carpeta de logs ' . $e->getMessage());
+        }
+    }
+
+     /**
+     * Returns the directory where the logs are stored.
+     * @return string
+     */
+    private function getLogDir(): string
+    {
+        return $this->logDir;
+    }
+
+    /**
+     * Generates a list of log files.
+     * @return array|null
+     */
+    private function setLogList(): array
+    {
+        $arr = array_diff(scandir($this->logDir), array('.', '..'));
+        foreach ($arr as $value) {
+            chmod($this->logDir . "/" . $value, 0660);
+            $logList[] = "<a href='{$this->logURL}/{$value}' download>{$value}</a>";
+        }
+        return $logList ?? null;
+    }
+
+    /**
+     * Gets the information of the last log file.
+     * @return array
+     */
+    private function getLastLog(): array
+    {
+        $files = glob($this->logDir . "/*.log");
+        if (!$files) {
+            return ["No existen Logs disponibles"];
+        }
+        $files = array_combine($files, array_map("filemtime", $files));
+        arsort($files);
+        $this->lastLog = key($files);
+        $logContent = @file_get_contents($this->lastLog) ?: null;
+        return [
+            'log_file' => basename($this->lastLog),
+            'log_weight' => $this->formatBytes($this->lastLog),
+            'log_regs_lines' => count(file($this->lastLog)),
+            'log_content' => $logContent
+        ];
+    }
+
+    /**
+     * Count the number of logs in the directory.
+     * @return array
+     */
+    private function setLogCount(): array
+    {
+        $logList = $this->setLogList();
+        $count = isset($logList) ? count($logList) : 0;
+        return ['log_count' => $count];
+    }
+
+    /**
+     * Gets a summary of the current configuration and logs.
+     * @return string
+     */
+    public function getResume(): bool|string
+    {
+        $result = array(
+            'log_dir' => $this->getLogDir(),
+            'logs_count' => $this->setLogCount(),
+            'logs_list' => $this->setLogList(),
+            'last_log' => $this->getLastLog(),
+        );
+        return json_encode($result);
+    }
+
+    /**
+     * Print DEBUG log
+     */
+    public function logDebug($msg)
+    {
+        if (self::LOG_DEBUG_ENABLED) {
```

Comment:
this is a const, why is used in an if statement if don't have a way to change it